_example/main.go

```diff
@@ -32,21 +32,24 @@ func main() {
 		Password: "password1",
 	}}
 
-	s, err := server.NewDefaultServer("tcp", "localhost:5123", auth, driver)
-	if err != nil {
+	if s, err := server.NewDefaultServer(server.Config{
+		Protocol: "tcp",
+		Address:  "localhost:5123",
+		Auth:     auth,
+	}, driver); err != nil {
 		panic(err)
+	} else {
+		s.Start()
 	}
-
-	s.Start()
 }
 
 func createTestDatabase() *mem.Database {
-	db := mem.NewDatabase("test")
+	db := mem.NewDatabase("test").(*mem.Database)
```

Comment:
why is the casting needed?

Reply:
because `mem.NewDatabase` returns `sql.Database` interface which does not have `AddTable` function.